src/comm/CMakeLists.txt

```diff
@@ -77,6 +77,14 @@ endif()
 
 target_compile_definitions(comm PUBLIC QGC_ENABLE_BLUETOOTH)
 
+if(NOT EXISTS ${CMAKE_SOURCE_DIR}/libs/mavlink/include/mavlink/v2.0/ardupilotmega)
```

Comment:
What is this for? In general QGC doesn't use ardupilotmega dialect. It uses "all" dialect.

Reply:
I don't know to be honest. It was in the qmake stuff

Reply:
I think you can get rid of it

Reply:
Should I remove its usage in the code too?

Reply:
Can you point me at that?

Reply:
[NO_ARDUPILOT_DIALECT.txt](https://github.com/mavlink/qgroundcontrol/files/14983023/NO_ARDUPILOT_DIALECT.txt)

Reply:
If you are talking about thee NO_ARDUPILOT_DIALECT defines. Those should be there. They allow you to create a custom build which only has a single firmware type. Thus reducing the size of your custom build. So those references in the source to ifdef stuff out should stay.

Reply:
There is a PX4 one as well to make an ArduPilot only build. It also reduces some of the complexity in the UI since it's simpler to support a single firmware type.

Reply:
So two things:
* As I said NO_ARDUPILOT_DIALECT is used to compile out all ArduPilot specific code. It shouldn't really have "DIALECT" in the name, which confuses things with mavlink dialects. It really NO_ARDUPILOT_SUPPORT.
* MAV_AUTOPILOT_ARDUPILOTMEGA is a firmware type from the mavlink spec: MAV_AUTOPILOT_ARDUPILOTMEGA. Is to determine what firmware type the vehicle is running

Reply:
In the past QGC used to use the ArduPilot mavlink dialect to build against instead of common. That's because at that time that dialect was a superset of what Ardupilot and PX4 used. So it made thing in QGC be able to talk mavlink with both firmwares. Also ages ago I decided to stop supporting Ardupilot specific mavlink stuff and just focus on generic across the board support because it was too much of a pain to implement things multiple ways. From the mavlink spec side this has been cleaned up such that the "all" dialect is now the one that all firmwares support. Hence the whole "DIALECT" aspect of "NO_ARDUPILOT_DIALECT" no longer makes any sense. Whereas in the past it made more sense.

Reply:
Wait then why should we get rid of the NO_ARDUPILOT_DIALECT compile definition? Or are you referring the check if it exists? Like it should be a cmake option instead of checking if the file exists?

Reply:
You should get rid of this check:
```
if(NOT EXISTS ${CMAKE_SOURCE_DIR}/libs/mavlink/include/mavlink/v2.0/ardupilotmega)
   target_compile_definitions(${PROJECT_NAME} PUBLIC NO_ARDUPILOT_DIALECT) # TODO: Make this QGC_NO_ARDUPILOT_DIALECT
endif()
```

You shouldn't get rid of NO_ARDUPILOT_DIALECT.